src/AWS.Deploy.CLI/App.cs

```diff
@@ -48,29 +50,48 @@ public async Task<int> Run(string[] args)
         private static void SetExecutionEnvironment(string[] args)
         {
             const string envName = "AWS_EXECUTION_ENV";
-            const string awsDotnetDeployCLI = "aws-dotnet-deploy-cli";
 
-            var assemblyVersion = typeof(Program).Assembly
-                .GetCustomAttributes(typeof(AssemblyInformationalVersionAttribute), false)
-                .FirstOrDefault()
-                as AssemblyInformationalVersionAttribute;
+            var toolVersion = GetToolVersion();
+
+            // The leading and trailing whitespaces are intentional
+            var userAgent = $" lib/aws-dotnet-deploy-cli#{toolVersion} ";
+            if (args?.Length > 0)
+            {
+                // The trailing whitespace is intentional
+                userAgent = $"{userAgent}md/cli-args#{args[0]} ";
+            }
+
 
             var envValue = new StringBuilder();
+            var existingValue = Environment.GetEnvironmentVariable(envName);
 
             // If there is an existing execution environment variable add this tool as a suffix.
-            if(!string.IsNullOrEmpty(Environment.GetEnvironmentVariable(envName)))
+            if (!string.IsNullOrEmpty(existingValue))
             {
-                envValue.Append($"{Environment.GetEnvironmentVariable(envName)}_");
+                envValue.Append(existingValue);
             }
 
-            envValue.Append($"{awsDotnetDeployCLI}_{assemblyVersion?.InformationalVersion}");
+            envValue.Append(userAgent);
 
-            if (args?.Length > 0)
+            Environment.SetEnvironmentVariable(envName, envValue.ToString());
+        }
+
+        private static string GetToolVersion()
+        {
+            var assembly = typeof(App).GetTypeInfo().Assembly;
+            var version = assembly.GetCustomAttribute<AssemblyFileVersionAttribute>()?.Version;
+            if (version is null)
             {
-                envValue.Append($"_{args[0]}");
+                return string.Empty;
             }
 
-            Environment.SetEnvironmentVariable(envName, envValue.ToString());
+            var versionParts = version.Split('.');
+            if (versionParts.Length == 4)
```

Comment:
Could you add a comment about why you're doing this?

Reply:
Done